fakeredis/_server.py

```diff
@@ -2547,7 +2547,7 @@ class FakeSelector(BaseSelector):
     def check_can_read(self, timeout):
         if self.sock.responses.qsize():
             return True
-        if timeout <= 0:
+        if timeout is None or timeout <= 0:
```

Comment:
This doesn't seem quite right: from what I can tell, `timeout=None` in redis is meant to mean "infinity", not "0".

Reply:
The test case I have is tricky, as it seems to host hang the thread